orchagent/nhgorch.cpp

```diff
@@ -0,0 +1,1386 @@
+        const auto& nhg_it = m_syncdNextHopGroups.find(index);
+
+        if (op == SET_COMMAND)
+        {
+            string ips;
+            string aliases;
+            string weights;
+            string mpls_nhs;
+
+            /* Get group's next hop IPs and aliases */
+            for (auto i : kfvFieldsValues(t))
+            {
+                if (fvField(i) == "nexthop")
+                    ips = fvValue(i);
+
+                if (fvField(i) == "ifname")
+                    aliases = fvValue(i);
+
+                if (fvField(i) == "weight")
+                    weights = fvValue(i);
+
+                if (fvField(i) == "mpls_nh")
+                    mpls_nhs = fvValue(i);
+            }
+
+            /* Split ips and alaises strings into vectors of tokens. */
+            vector<string> ipv = tokenize(ips, ',');
+            vector<string> alsv = tokenize(aliases, ',');
+            vector<string> mpls_nhv = tokenize(mpls_nhs, ',');
+
+            /* Create the next hop group key. */
+            string nhg_str;
+
+            for (uint32_t i = 0; i < ipv.size(); i++)
+            {
+                if (i) nhg_str += NHG_DELIMITER;
+                if (!mpls_nhv.empty() && mpls_nhv[i] != "na")
+                {
+                    nhg_str += mpls_nhv[i] + LABELSTACK_DELIMITER;
+                }
+                nhg_str += ipv[i] + NH_DELIMITER + alsv[i];
+            }
+
+            NextHopGroupKey nhg_key = NextHopGroupKey(nhg_str, weights);
+
+            /* If the group does not exist, create one. */
+            if (nhg_it == m_syncdNextHopGroups.end())
+            {
+                SWSS_LOG_NOTICE("Adding next hop group %s with %s",
+                                index.c_str(),
+                                nhg_str.c_str());
+                /*
+                * If we'd have to create a SAI object for the group, and we
+                * already reached the limit, we're going to create a temporary
+                * group, represented by one of it's NH only until we have
+                * enough resources to sync the whole group.  The item is going
+                * to be kept in the sync list so we keep trying to create the
+                * actual group when there are enough resources.
+                */
+                if ((nhg_key.getSize() > 1) &&
+                    (NextHopGroup::getCount() >= m_maxNhgCount))
+                {
+                    SWSS_LOG_WARN("Next hop group count reached its limit.");
+
+                    try
+                    {
+                        auto nhg = std::make_unique<NextHopGroup>(
+                                                    createTempNhg(nhg_key));
+                        SWSS_LOG_NOTICE("Adding temp next hop group with %s",
+                                        nhg->to_string().c_str());
+                        if (nhg->sync())
+                        {
+                            SWSS_LOG_INFO("Temporary NHG successfully synced");
```

Comment:
remove this log. In general logs for successful cases are less useful than logs for failure cases.

Reply:
Done